ctrd/image.go

```diff
@@ -94,14 +94,24 @@ func (c *Client) PullImage(ctx context.Context, ref string, stream *jsonstream.J
 
 	// cancel fetch progress befor handle error.
 	cancelProgress()
+	defer stream.Close()
+
+	// wait fetch progress to finish.
+	<-wait
 
 	if err != nil {
+		// Send Error information to client through stream
+		var msgList []ProgressInfo
+		errMsg := ProgressInfo{
+			Code:         http.StatusInternalServerError,
+			ErrorMessage: err.Error(),
+		}
+		msgList = append(msgList, errMsg)
```

Comment:
s/msgList/messages/g , the variable name does not need to contain it's type.

Others, We can combine those codes into one line.
messages := []ProgressInfo{
{
Code: xxxx,
ErrorMessage: xxxx,
}
}